### PR TITLE
toolkit: containerized-build: honor $SPECS_DIR instead of hard-coding

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Honor `$SPECS_DIR` instead of hardcoding the value.

### Test1: Nothing provided on the command line
```bash
sudo make containerized-rpmbuild SRPM_PACK_LIST=vim
# ... [snip] ...
* Mount points:
/home/mfrw/mariner-org/cve/cve-ext/build/container-build' -> '/usr/src/mariner/BUILD
/home/mfrw/mariner-org/cve/cve-ext/build/container-buildroot' -> '/usr/src/mariner/BUILDROOT
/home/mfrw/mariner-org/cve/cve-ext/out/RPMS' -> '/mnt/RPMS
/home/mfrw/mariner-org/cve/cve-ext/build/containerized-rpmbuild/tmp' -> '/mariner_setup_dir
/home/mfrw/mariner-org/cve/cve-ext/build/INTERMEDIATE_SRPMS' -> '/mnt/INTERMEDIATE_SRPMS
/home/mfrw/mariner-org/cve/cve-ext/SPECS' -> '/usr/src/mariner/SPECS

```

### Test2: Relative PATHS
```bash
sudo make containerized-rpmbuild SRPM_PACK_LIST=vim SPECS_DIR=../SPECS
# ... [snip] ...
* Mount points:
/home/mfrw/mariner-org/cve/cve-ext/build/container-build' -> '/usr/src/mariner/BUILD
/home/mfrw/mariner-org/cve/cve-ext/build/container-buildroot' -> '/usr/src/mariner/BUILDROOT
/home/mfrw/mariner-org/cve/cve-ext/out/RPMS' -> '/mnt/RPMS
/home/mfrw/mariner-org/cve/cve-ext/build/containerized-rpmbuild/tmp' -> '/mariner_setup_dir
/home/mfrw/mariner-org/cve/cve-ext/build/INTERMEDIATE_SRPMS' -> '/mnt/INTERMEDIATE_SRPMS
/home/mfrw/mariner-org/cve/cve-ext/SPECS' -> '/usr/src/mariner/SPECS
```

### Test3: Absolute PATHS
```bash
sudo make containerized-rpmbuild SRPM_PACK_LIST=vim SPECS_DIR=/home/mfrw/mariner-org/CBL-Mariner/SPECS
# ... [snip] ...
* Mount points:
/home/mfrw/mariner-org/cve/cve-ext/build/container-build' -> '/usr/src/mariner/BUILD
/home/mfrw/mariner-org/cve/cve-ext/build/container-buildroot' -> '/usr/src/mariner/BUILDROOT
/home/mfrw/mariner-org/cve/cve-ext/out/RPMS' -> '/mnt/RPMS
/home/mfrw/mariner-org/cve/cve-ext/build/containerized-rpmbuild/tmp' -> '/mariner_setup_dir
/home/mfrw/mariner-org/cve/cve-ext/build/INTERMEDIATE_SRPMS' -> '/mnt/INTERMEDIATE_SRPMS
/home/mfrw/mariner-org/cve/cve-ext/SPECS' -> '/usr/src/mariner/SPECS
```

### Test4: EXTENDED Relative
```bash
sudo make containerized-rpmbuild SRPM_PACK_LIST=wireshark SPECS_DIR=../SPECS-EXTENDED/
# ... [snip] ...
* Mount points:
* Mount points:
/home/mfrw/mariner-org/cve/cve-ext/build/container-build' -> '/usr/src/mariner/BUILD
/home/mfrw/mariner-org/cve/cve-ext/build/container-buildroot' -> '/usr/src/mariner/BUILDROOT
/home/mfrw/mariner-org/cve/cve-ext/out/RPMS' -> '/mnt/RPMS
/home/mfrw/mariner-org/cve/cve-ext/build/containerized-rpmbuild/tmp' -> '/mariner_setup_dir
/home/mfrw/mariner-org/cve/cve-ext/build/INTERMEDIATE_SRPMS' -> '/mnt/INTERMEDIATE_SRPMS
/home/mfrw/mariner-org/cve/cve-ext/SPECS-EXTENDED' -> '/usr/src/mariner/SPECS
```

### Test4: EXTENDED Absolute PATHS
```bash
sudo make containerized-rpmbuild SRPM_PACK_LIST=wireshark SPECS_DIR=/home/mfrw/mariner-org/CBL-Mariner/SPECS-EXTENDED
# ... [snip] ...
* Mount points:
* Mount points:
/home/mfrw/mariner-org/cve/cve-ext/build/container-build' -> '/usr/src/mariner/BUILD
/home/mfrw/mariner-org/cve/cve-ext/build/container-buildroot' -> '/usr/src/mariner/BUILDROOT
/home/mfrw/mariner-org/cve/cve-ext/out/RPMS' -> '/mnt/RPMS
/home/mfrw/mariner-org/cve/cve-ext/build/containerized-rpmbuild/tmp' -> '/mariner_setup_dir
/home/mfrw/mariner-org/cve/cve-ext/build/INTERMEDIATE_SRPMS' -> '/mnt/INTERMEDIATE_SRPMS
/home/mfrw/mariner-org/cve/cve-ext/SPECS-EXTENDED' -> '/usr/src/mariner/SPECS
```

###### Change Log  

- Honor value of `$SPECS_DIR` instead of only mounting `SPECS`

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- Local Build and test

### CODE REVIEW

**Review by neha170 — COMMENTED** [review]
Thanks for adding this, very useful and important edge case. Couple things,
- Could you make it into an optional argument, like the others. By default, it will point to the specs folder, but if provided, it will mount the user provided specs folder.
- Please update the documentation as well (help function)
- The reason it is not able to find SPECS-EXTENDED is because we run the make command from toolkit, but the script resides at toolkit/scripts/containerized-build. Providing a full path to MOUNTS works in general. In this case however, it refuses to mount two locations to SPECS/ inside the container.
- I have made change to use realpath for mounting extra mounts provided by user. I'll publish a PR soon, along with other suggested changes

**Comment by mfrw** [comment]
> Thanks for adding this, very useful and important edge case. Couple things,
> 
> * Could you make it into an optional argument, like the others. By default, it will point to the specs folder, but if provided, it will mount the user provided specs folder.
> * Please update the documentation as well (help function)
> * The reason it is not able to find SPECS-EXTENDED is because we run the make command from toolkit, but the script resides at toolkit/scripts/containerized-build. Providing a full path to MOUNTS works in general. In this case however, it refuses to mount two locations to SPECS/ inside the container.
> * I have made change to use realpath for mounting extra mounts provided by user. I'll publish a PR soon, along with other suggested changes

Ahha, you replied in the midst of me updating the comments. 
The default is `SPECS` but if we provide annything else, it will take that.
As of now, it does not behave exactly I want it to behave - so I will mark it as draft :)

File: toolkit/scripts/containerized-build/create_container_build.sh [hunk 1]
```diff
@@ -29,6 +29,8 @@ Starts a docker container with the specified version of mariner.
```
Comment (neha170) [inline toolkit/scripts/containerized-build/create_container_build.sh:29]: Please add [SPECS_DIR=/path/to/SPECS_DIR/to/use] to Usage on #26

**Review by neha170 — APPROVED** [review]
LGTM, just a minor comment

**Review by dmcilvaney — APPROVED** [review]

File: toolkit/scripts/containerized-build/create_container_build.sh (hunk 1 above)
Comment (mfrw) [reply toolkit/scripts/containerized-build/create_container_build.sh:29]: Sure :) Done
